File: integration-tests/src/test/java/pl/grzeslowski/capybara/test/BoolLogicTest.java
package pl.grzeslowski.capybara.test;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class BoolLogicTest {
    @Test
    void andLogic() {
        assertThat(BoolLogic.andLogic(true, true)).isTrue();
        assertThat(BoolLogic.andLogic(true, false)).isFalse();
        assertThat(BoolLogic.andLogic(false, true)).isFalse();
        assertThat(BoolLogic.andLogic(false, false)).isFalse();
    }

    @Test
    void orLogic() {
        assertThat(BoolLogic.orLogic(true, true)).isTrue();
        assertThat(BoolLogic.orLogic(true, false)).isTrue();
        assertThat(BoolLogic.orLogic(false, true)).isTrue();
        assertThat(BoolLogic.orLogic(false, false)).isFalse();
    }

    @Test
    void equalLogic() {
        assertThat(BoolLogic.equalLogic(true, true)).isTrue();
        assertThat(BoolLogic.equalLogic(false, false)).isTrue();
        assertThat(BoolLogic.equalLogic(true, false)).isFalse();
        assertThat(BoolLogic.equalLogic(false, true)).isFalse();
    }

    @Test
    void notEqualLogic() {
        assertThat(BoolLogic.notEqualLogic(true, true)).isFalse();
        assertThat(BoolLogic.notEqualLogic(false, false)).isFalse();
        assertThat(BoolLogic.notEqualLogic(true, false)).isTrue();
        assertThat(BoolLogic.notEqualLogic(false, true)).isTrue();
    }

    @Test
    void notLogic() {
        assertThat(BoolLogic.notLogic(true)).isFalse();
        assertThat(BoolLogic.notLogic(false)).isTrue();
    }

    @Test
    void complexLogic() {
        assertThat(BoolLogic.complexLogic(true, true)).isTrue();
        assertThat(BoolLogic.complexLogic(true, false)).isFalse();
        assertThat(BoolLogic.complexLogic(false, true)).isTrue();
        assertThat(BoolLogic.complexLogic(false, false)).isTrue();
    }
}

File: capy/src/e2e-cfun/js/dev/capylang/test/BoolLogic.test.js
const test = require('node:test');
const { assert, generatedModule } = require('./_support');

test('BoolLogic', () => {
    const boolLogic = generatedModule('dev/capylang/test/BoolLogic.js');
    assert.equal(boolLogic.andLogic(true, true), true);
    assert.equal(boolLogic.andLogic(true, false), false);
    assert.equal(boolLogic.andLogic(false, true), false);
    assert.equal(boolLogic.orLogic(false, true), true);
    assert.equal(boolLogic.orLogic(false, false), false);
    assert.equal(boolLogic.equalLogic(true, true), true);
    assert.equal(boolLogic.notEqualLogic(true, false), true);
    assert.equal(boolLogic.notLogic(false), true);
    assert.equal(boolLogic.notFieldAccess(new boolLogic.BoolBox({ failed: true })), false);
    assert.equal(boolLogic.notMethodCall(new boolLogic.BoolBox({ failed: false })), true);
    assert.equal(boolLogic.complexLogic(false, true), true);
    assert.equal(boolLogic.byteToBool(0), false);
    assert.equal(boolLogic.intToBool(2), true);
    assert.equal(boolLogic.stringToBool(''), false);
    assert.equal(boolLogic.listToBool([1]), true);
    assert.equal(boolLogic.setToBool(new Set()), false);
    assert.equal(boolLogic.dictToBool(new Map([['a', 1]])), true);
    assert.equal(boolLogic.andInt(2, 1), true);
    assert.equal(boolLogic.orString('', ''), false);
    assert.equal(boolLogic.notList([]), true);
});
